src/controller/python/chip/clusters/Command.py

```diff
@@ -117,20 +128,91 @@ def handleError(self, status: Status, chipError: PyChipError):
             self._handleError, status, chipError, None
         )
 
+    def handleDone(self):
+        ctypes.pythonapi.Py_DecRef(ctypes.py_object(self))
+
+
+class AsyncBatchCommandsTransaction:
+    def __init__(self, future: Future, eventLoop, expectTypes: List[Type]):
+        self._event_loop = eventLoop
+        self._future = future
+        self._expect_types = expectTypes
+        default_im_failure = chip.interaction_model.InteractionModelError(
+            chip.interaction_model.Status.Failure)
+        self._responses = [default_im_failure] * len(expectTypes)
+
+    def _handleResponse(self, path: CommandPath, index: int, status: Status, response: bytes):
+        if index > len(self._responses):
+            self._handleError(status, 0, IndexError(f"CommandSenderCallback has given us an unexpected index value {index}"))
+            return
+
+        if (len(response) == 0):
+            self._responses[index] = None
+        else:
+            # If a type hasn't been assigned, let's auto-deduce it.
+            if (self._expect_types[index] is None):
```

Comment:
Is the expectation that this is provided for things like manufacturer specific commands?

Reply:
That is my understanding. Otherwise as far as my testing goes we likely could always auto-deduce everything else I have thrown at it. IIRC yamltest parse using chip-repl runner always replied on auto-deduction.